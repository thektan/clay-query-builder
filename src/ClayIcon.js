import React from 'react';
import {PropTypes} from 'prop-types';
import ThemeContext from './ThemeContext';

class ClayIcon extends React.Component {
    static contextType = ThemeContext;

    render() {
        const {className, iconName} = this.props;

        return (
            <svg aria-hidden="true" className={`lexicon-icon ${className}`} viewBox="0 0 512 512">
                <use xlinkHref={ `${this.context.spriteMapPath}#${iconName}` }></use>
            </svg>
        );
    }
}

ClayIcon.propTypes = {
    className: PropTypes.string,
    iconName: PropTypes.string,
}

export default ClayIcon;