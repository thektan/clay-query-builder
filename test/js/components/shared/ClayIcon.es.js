import React from 'react';
import ClayIcon from 'components/shared/ClayIcon.es';
import {cleanup, render} from 'react-testing-library';

describe(
	'ClayIcon',
	() => {
		afterEach(cleanup);

		it(
			'should render',
			() => {
				const {asFragment} = render(
					<ClayIcon iconName="times" />
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);
	}
);