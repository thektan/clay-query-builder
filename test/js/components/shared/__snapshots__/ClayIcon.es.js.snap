// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayIcon should render 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="lexicon-icon lexicon-icon-times"
    viewBox="0 0 512 512"
  >
    <use
      xlink:href="#times"
    />
  </svg>
</DocumentFragment>
`;
